### PR TITLE
Sami/permission checker

### PR DESCRIPTION
**Pull request by MSamiTariq**

### CODE REVIEW

File: contracts/marketupdates/MarketAdminPermissionChecker.sol [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    error AlreadyUnPaused();
+
+    /**
+     * @notice Construct a new MarketAdminPermissionChecker contract.
+     * Not adding any checks for zero address as it may be a deliberate choice to assign the zero address i.e. keep the
+     * market updates disabled.
+     * @param _marketAdmin The address of the market admin.
+     * @param _marketAdminPauseGuardian The address of the market admin pause guardian.
+     */
+    constructor(address _marketAdmin, address _marketAdminPauseGuardian) {
```
Comment (DawoodMehmood) [inline contracts/marketupdates/MarketAdminPermissionChecker.sol:34]: should be marketAdmin_ and marketAdminPauseGuardian_ as per practice

File: contracts/vendor/access/Ownable.sol [hunk 2]
```diff
@@ -14,7 +15,7 @@ import "../utils/Context.sol";
  * can later be changed with {transferOwnership}.
  *
  * This module is used through inheritance. It will make available the modifier
- * `onlyOwner`, which can be applied to your functions to restrict their use to
+ * `onlyOwner`, which can be applied to your funwctions to restrict their use to
```
Comment (DawoodMehmood) [inline contracts/vendor/access/Ownable.sol:18]: wrong spell and there should be no change in this contract

File: contracts/marketupdates/MarketAdminPermissionChecker.sol [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+     */
+    function unpauseMarketAdmin() external onlyOwner {
+        if (!marketAdminPaused) revert AlreadyUnPaused();
+        marketAdminPaused = false;
+        emit MarketAdminPaused(msg.sender, false);
+    }
+
+    function checkUpdatePermission(address callerAddress) external view {   // update the msg.sender
+        if (callerAddress != marketAdmin) revert Unauthorized();
+        if (marketAdminPaused) revert MarketAdminIsPaused();
```
Comment (DawoodMehmood) [inline contracts/marketupdates/MarketAdminPermissionChecker.sol:92]: for the same condition above at line #73 the event emitted is AlreadyPaused() but here the event is MarketAdminIsPaused()

Comment (MSamiTariq) [reply contracts/marketupdates/MarketAdminPermissionChecker.sol:90]: I believe the rationale is that, in the earlier case, we were attempting to pause a market admin that was already paused, so the `AlreadyPaused` error was appropriate. However, in this case, the check is related to permissions, so the AlreadyPaused error doesn’t seem applicable.

File: test/marketupdates/market-update-deployment-test.ts [hunk 4]
```diff
@@ -174,21 +199,25 @@ describe('MarketUpdateDeployment', function() {
...
-      )]
+      [
+        ethers.utils.defaultAbiCoder.encode(
+          ['address', 'address'],
+          [configuratorProxyContract.address, proxyAdminNew.address]
+        ),
+      ]
     );
 
     //
```
Comment (DawoodMehmood) [inline test/marketupdates/market-update-deployment-test.ts:223]: empty comment

**Pull request merged**